assets/tests/helpers/svgIcon.test.tsx

```diff
@@ -0,0 +1,76 @@
+import { render, screen } from "@testing-library/react"
+import React from "react"
+import { SvgIcon, svgIcon } from "../../src/helpers/svgIcon"
+import "@testing-library/jest-dom"
+
+describe("<SvgIcon/>", () => {
+  test("renders the given svg content in a span element with a class name", () => {
+    const className = "test-class-name"
+    const svgText = "svg text"
+
+    const result = render(
+      <SvgIcon
+        svgText={`<svg>${svgText}</svg>`}
+        className={className}
+      ></SvgIcon>
+    ).asFragment()
+
+    expect(result).toEqual(
+      render(
+        <span className={className}>
+          <svg>{svgText}</svg>
+        </span>
+      ).asFragment()
+    )
+  })
+})
+
+describe("svgIcon(text) -> (props)", () => {
+  describe("first function returns react element which renders `svgText` and passed props", () => {
+    test("when called via function call", () => {
+      const className = "test-class-name"
+      const svgText = "<svg><title>hello test</title></svg>"
+
+      const TestSvgTextElement = svgIcon(svgText)
+      const result = render(
+        TestSvgTextElement({ className, role: "img" })
+      ).asFragment()
+
+      expect(result).toEqual(
+        render(
+          <span className={className} role="img">
+            <svg>
+              <title>hello test</title>
+            </svg>
+          </span>
+        ).asFragment()
+      )
+    })
+
+    // test("when element is created from svg text, element should be in the accessibility tree from specified props", () => {
+    test("when used via JSX", () => {
```

Comment:
question (non-blocking): were these intended to be two separate tests?

Reply:
I'm gonna leave it given no objections before merge, we can always remove it or merge them later?